src/compiler/Stage/Desugar.elm

```diff
@@ -146,6 +149,13 @@ desugarLet recurse bindings body =
         (recurse body)
 
 
+desugarList : (Frontend.Expr -> Result DesugarError Canonical.Expr) -> List Frontend.Expr -> Result DesugarError Canonical.Expr
+desugarList recurse list =
+    List.map recurse list
+        |> List.foldl (Result.map2 (::)) (Ok [])
```

Comment:
some `Debug.log`s:
```
source:           [ 1, 50, 999 ]
after parsing:    List [Literal (Int 1),Literal (Int 50),Literal (Int 999)]
after desugaring: List [Literal (Int 999),Literal (Int 50),Literal (Int 1)]
after typing:     (List [(Literal (Int 999),Int),(Literal (Int 50),Int),(Literal (Int 1),Int)],Var 4)
after emit:       [999, 50, 1]
```

I believe the `foldr` was correct. The comment below was just a question of why is the `foldr` there, didn't want to indicate that it's wrong ... :) sorry for confusion!